sdk/endpoints/online/managed/managed-identities/online-endpoints-managed-identity-uai.ipynb

```diff
@@ -0,0 +1,849 @@
+    "    environment=Environment(\n",
+    "        conda_file=\"../../model-1/environment/conda.yml\",\n",
+    "        image=\"mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04:20210727.v1\",\n",
+    "    ),\n",
+    "    instance_type=\"Standard_DS2_v2\",\n",
+    "    instance_count=1,\n",
+    "    environment_variables={\n",
+    "        \"STORAGE_ACCOUNT_NAME\": storage_account_name,\n",
+    "        \"STORAGE_CONTAINER_NAME\": storage_container_name,\n",
+    "        \"FILE_NAME\": file_name,\n",
+    "        # We will update this after creating an identity\n",
+    "        \"UAI_CLIENT_ID\": \"uai_client_id_place_holder\",\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Create the managed identity"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.1 Get a handle to the `ManagedServiceIdentityClient`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "name": "3-get-handle"
+   },
+   "outputs": [],
+   "source": [
+    "from azure.identity import DefaultAzureCredential\n",
+    "from azure.mgmt.msi import ManagedServiceIdentityClient\n",
+    "from azure.mgmt.msi.models import Identity\n",
+    "\n",
+    "credential = AzureCliCredential()\n",
+    "msi_client = ManagedServiceIdentityClient(\n",
+    "    subscription_id=subscription_id,\n",
+    "    credential=credential,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.2 Create the identity"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name": "3-create-identity"
+   },
+   "outputs": [],
+   "source": [
+    "msi_client.user_assigned_identities.create_or_update(\n",
+    "    resource_group_name=resource_group,\n",
+    "    resource_name=uai_name,\n",
+    "    parameters=Identity(location=workspace_location),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.3 Retrieve the identity object"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name": "3-retrieve-identity-object"
+   },
+   "outputs": [],
+   "source": [
+    "uai_identity = msi_client.user_assigned_identities.get(\n",
+    "    resource_group_name=resource_group,\n",
+    "    resource_name=uai_name,\n",
+    ")\n",
+    "uai_identity.as_dict()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Create storage account and container"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4.1 Get a handle to the `StorageManagementclient`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name": "4-get-handle"
+   },
+   "outputs": [],
+   "source": [
+    "from azure.identity import DefaultAzureCredential, AzureCliCredential\n",
```

Comment:
Do we need to import this again?
`from azure.identity import DefaultAzureCredential, AzureCliCredential`

Reply:
Resolved, thanks